### PR TITLE
change timestamp type to LocalDateTime

### LINKED ISSUE #100: Data Type mapping compliance with R2DBC spec
We need to ensure compliance with the data type mapping as [defined in the spec](https://r2dbc.io/spec/0.8.0.M8/spec/html/#datatypes), and recently amended in https://github.com/r2dbc/r2dbc-spi/pull/100.

If we are unable to comply with any of it, it should be documented.

### PR DESCRIPTION
**Pull request by dmitry-s**
fixes #100

### CODE REVIEW

File: src/test/java/com/google/cloud/spanner/r2dbc/codecs/DefaultCodecsTest.java [hunk 1]
```diff
@@ -97,13 +115,16 @@
   @Parameter(2)
   public Type valueType;
 
+  @Parameter(3)
+  public Object decodedVal;
```
Comment (meltsufin) [inline src/test/java/com/google/cloud/spanner/r2dbc/codecs/DefaultCodecsTest.java]: Can you avoid this?
There should be a way to check if two timestamps are equivalent regardless of their timezone.

Comment (dmitry-s) [reply src/test/java/com/google/cloud/spanner/r2dbc/codecs/DefaultCodecsTest.java]: good idea

**Comment by ChengyuanZhao** [comment]
@meltsufin @dmitry-s the only difference between the two is that `ZonedDateTime` knows about things like daylight savings time. A "zone" is an "offset" plus daylight savings time adjustment. 

It makes no difference to us because Spanner can store neither an offset nor a time zone. In either case we can only create an object using either 0 offset or "UTC" zone. The UTC time zone has no DST adjustments, so it is the same as 0 offset always.

File: README.md [hunk 2]
```diff
@@ -168,7 +168,7 @@ Cloud Spanner R2DBC Driver supports the following types:
 |`INT64`         |`java.lang.Long`     |
 |`INT64`         |`java.lang.Integer`  |
 |`STRING`        |`java.lang.String`   |
-|`TIMESTAMP`     |`java.sql.Timestamp` |
+|`TIMESTAMP`     |`java.time.ZonedDateTime` |
```
Comment (meltsufin) [inline README.md]: Still says zoned.

**Pull request merged**